[FEAT] Configurable character limit on live chat

ogustavo-pereira opened the pull request:
## Proposed changes

Implementing a new character limit field in the LiveChat Aparence.

## Issue(s)

#17451

## How to test or reproduce

1. Open Livechat Aparence
2. Edit Characther Limit input
3. Save
4. reload the page and check if the information has been saved

If you want test with Livechat, you will need this [feature](https://github.com/RocketChat/Rocket.Chat.Livechat/pull/443)

## Screenshots

![image](https://user-images.githubusercontent.com/24482087/87487216-75210d00-c613-11ea-913a-68c5c023f636.png)

## Types of changes
- [x] New feature (non-breaking change which adds functionality)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [ ] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat)
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

## Changelog

additional setting in Admin panel for setting a character limit on livechat.

## Further comments

graywolf336 committed:
Submit a payload to the release service when a release happens (#17775)

graywolf336 committed:
Fix invalid develop payload to release service (#17799)

The payload sent to the release service contains an invalid reference to the tag/version and wasn't working when building develop branch.

graywolf336 committed:
Fix invalid develop payload to release service (#17799)

The payload sent to the release service contains an invalid reference to the tag/version and wasn't working when building develop branch.

d-gubert committed:
Update Apps-Engine version (#17804)

d-gubert committed:
Update Apps-Engine version (#17804)

d-gubert committed:
[FIX] Error when re-installing an App (#17789)

* Fix wrong method call on app installation

* Update subscription check to take cancelled state into account

d-gubert committed:
[FIX] Error when re-installing an App (#17789)

* Fix wrong method call on app installation

* Update subscription check to take cancelled state into account

sampaiodiego committed:
[FIX] Email link "go to message" being incorrectly escaped (#17803)

sampaiodiego committed:
[FIX] Email link "go to message" being incorrectly escaped (#17803)

gabriellsh committed:
[FIX] Link preview containing HTML encoded chars (#16512)

d-gubert committed:
Add Apps-Engine to Engine Versions on History (#17810)

graywolf336 committed:
Fix the update check not working (#17809)

Co-authored-by: Aaron Ogle <geekgonecrazy@users.noreply.github.com>

graywolf336 committed:
Fix the update check not working (#17809)

Co-authored-by: Aaron Ogle <geekgonecrazy@users.noreply.github.com>

alansikora committed:
[IMPROVE][Federation] Add support for _tcp and protocol DNS entries (#17818)

ishriom53tyagi committed:
[FIX] When the message is too long declining to send as an attachment does not restore the content into the composer (#16332)

Co-authored-by: Danish <danish@kitchen.mshome.net>
Co-authored-by: Rodrigo Nascimento <rodrigoknascimento@gmail.com>
Co-authored-by: Guilherme Gazzo <guilhermegazzo@gmail.com>

pierre-lehnen-rc committed:
[NEW] Assign oldest active user as owner when deleting last room owner (#16088)

Co-authored-by: Diego Sampaio <chinello@gmail.com>

Commit:
[REGRESSION] Omnichannel visitor forward was applying wrong restrictions (#17826)

* return empty object if there is no callback

* Improve department forward restrictions

Commit:
[REGRESSION] Omnichannel visitor forward was applying wrong restrictions (#17826)

* return empty object if there is no callback

* Improve department forward restrictions

dudizilla committed:
[FIX] Administration User page blank opening users without email (#17836)

dudizilla committed:
[FIX] Administration User page blank opening users without email (#17836)

rodrigok committed:
Create codeql-analysis.yml

lpilz committed:
[FIX] Slack importer settings object (#17776)

lpilz committed:
[FIX] Slack importer settings object (#17776)

tassoevan committed:
Refactor components and views to Storybook compatibility (#17800)

cking-vonix committed:
[FIX] Logic for room type was inverted in edit panel (#17851) (#17853)

cking-vonix committed:
[FIX] Logic for room type was inverted in edit panel (#17851) (#17853)

renatobecker committed:
[FIX] Omnichannel message link is broken in email notifications (#17843)

renatobecker committed:
[FIX] Omnichannel message link is broken in email notifications (#17843)

pierre-lehnen-rc committed:
[FIX] Importers progress sending too much update events to clients (#17857)

* Delay reporting the import progress to the frontend so it doesn't overwhelm the connection

* Increased time to 250ms

d-gubert committed:
[FIX] Apps room events losing data (#17827)

d-gubert committed:
[FIX] Apps room events losing data (#17827)

pierre-lehnen-rc committed:
[FIX] SAML LogoutRequest sending wrong NameID (#17860)

pierre-lehnen-rc committed:
[FIX] SAML LogoutRequest sending wrong NameID (#17860)

MartinSchoeler committed:
[FIX] Reorder hljs (#17854)

sampaiodiego committed:
Bump version to 3.3.1

pierre-lehnen-rc committed:
[NEW] Blocked Media Types setting (#17617)

sampaiodiego committed:
Merge pull request #17865 from RocketChat/release-3.3.1

Release 3.3.1

sampaiodiego committed:
Bump version to 3.3.2

sampaiodiego committed:
Merge pull request #17870 from RocketChat/release-3.3.2

Release 3.3.2

sampaiodiego committed:
Always initialize CIRCLE_BRANCH env var on CI (#17874)

sampaiodiego committed:
Always initialize CIRCLE_BRANCH env var on CI (#17874)

sampaiodiego committed:
Bump version to 3.3.3

sampaiodiego committed:
Merge pull request #17875 from RocketChat/release-3.3.3

Release 3.3.3

gabriellsh committed:
[IMPROVE] Rewrite admin sidebar in React (#17801)

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>
Co-authored-by: Guilherme Gazzo <guilhermegazzo@gmail.com>

tassoevan committed:
Change some components' location (#17893)

pierre-lehnen-rc committed:
[FIX] Some Login Buttons disappear after refreshing OAuth Services (#17808)

Co-authored-by: Diego Sampaio <chinello@gmail.com>

ggazzo committed:
Wrap components with React.memo (#17899)

dudizilla committed:
[FIX] Duplicated placeholderaceholder  (#17898)

tassoevan committed:
[IMPROVE] Rewrite Federation Dashboard (#17900)

Sing-Li committed:
correct outdated dev start info

jazztickets committed:
[IMPROVE] Split NOTIFICATIONS_SCHEDULE_DELAY into three separate variables (#17669)

sampaiodiego committed:
[IMPROVE] Don't send emails to online users and remove delay when away/idle (#17907)

* Do not send email if user online

* Remove delay when away

sampaiodiego committed:
Remove unused accounts-js integration (#17921)

pierre-lehnen-rc committed:
[FIX] Undesirable message updates after user saving profile (#17930)

rodrigok committed:
[IMPROVE][Performance] Add new database indexes to improve data query performance (#17839)

Co-authored-by: Diego Sampaio <chinello@gmail.com>

sampaiodiego committed:
Update Dockerfile to not depend on custom base image (#17802)

* Remove dependency from Base Docker image

* Update preview entrypoint for Mongo 4.2

* Use node base image

sampaiodiego committed:
[IMPROVE] Change default upload settings to only block SVG files (#17933)

engelgabriel committed:
LingoHub based on develop (#17796)

* LingoHub Update :rocket:

Manual push by LingoHub User: Diego Sampaio.
Project: Rocket.Chat

Made with :heart: by https://lingohub.com

* Remove old setting i18n

Co-authored-by: Diego Sampaio <chinello@gmail.com>

sampaiodiego committed:
[IMPROVE] Add rate limiter to UiKit endpoints (#17859)

* Fix unused variables

* Add rate limiter to UiKit endpoints

* Uses 60 times default rate limit values

sampaiodiego committed:
[IMPROVE] Use REST for DDP calls by default (#17934)

sampaiodiego committed:
Do not build Docker image for fork PRs (#17370)

mrsimpson committed:
Chatpal: limit results to current room (#17718)

ashwaniYDV committed:
[FIX] Missing pinned icon indicator for messages pinned (#16448)

mohamedar97 committed:
[FIX] Close the user info context panel does not navigate back to the user's list (#14085)

antkaz committed:
[FIX] Not possible to translate the label of custom fields in user's Info (#15595)

vegeta999 committed:
[FIX] Message action popup doesn't adjust itself on screen resize (#16508)

ashwaniYDV committed:
[NEW][API] New endpoints to manage User Custom Status `custom-user-status.create`, custom-user-status.delete` and `custom-user-status.update` (#16550)

gabriellsh committed:
[FIX] Links being escaped twice leading to visible encoded characters (#16481)

Karting06 committed:
[FIX] Outgoing webhook: Excessive spacing between trigger words (#17830)

vegeta999 committed:
[FIX] Profile save button not activates properly when changing the username field (#16541)

sampaiodiego committed:
[FIX] StreamCast stream to server only streamers (#17942)

gabriellsh committed:
[IMPROVE] React hooks lint rules (#17941)

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>

tassoevan committed:
[IMPROVE] Performance editing Admin settings (#17916)

ggazzo committed:
[FIX] Hide system message add/remove owner  (#17938)

ggazzo committed:
[NEW][ENTERPRISE] Download engagement data (#17920)

tassoevan committed:
Regression: Infinite loop in CodeSettingInput (#17949)

toshokan committed:
[FIX] Typo "coorosponding" (#17840)

sampaiodiego committed:
Add Apps to control GitHub issues (#17807)

ggazzo committed:
[FIX] Discussion List paddings (#17955)

Cleod9 committed:
[FIX] Added explicit server oembed provider for Twitter (#17954)

pierre-lehnen-rc committed:
[FIX] ReadOnly Rooms permission checks (#17709)

Karting06 committed:
[IMPROVE] Allow webhook message to respond in thread (#17863)

dependabot[bot] committed:
Bump websocket-extensions from 0.1.3 to 0.1.4 (#17837)

thomas-mc-work committed:
LDAP typo  (#17835)

Commit:
[FIX] double slashes in avatar url (#17739)

onurtemiz committed:
Fix typo on Contributing.md (#17769)

pierre-lehnen-rc committed:
[IMPROVE] User avatar cache invalidation (#17925)

Co-authored-by: Diego Sampaio <chinello@gmail.com>

sampaiodiego committed:
[FIX] Video conferences being started by users without permission (#17948)

sampaiodiego committed:
Update stale bot to v3 and run every 6 hours (#17958)

nitinkumartiwari committed:
[IMPROVE] Refactor Omnichannel Past Chats List (#17346)

* Add all the features

* css improved

* changed some css

* Code and CSS Improvement

* Code and CSS Improvement

* some changes

* changes made

* Api modified

* css changes with Api modification

* chnages made on 28April

* changes made on 1may

* changes made on 6th may

* css chang

* css chang

* css chang

* css chang

* css chang

* css chang

* css chang

* changes made on 7th may

* Update visitors.js

* Fix indentations.

* Fix css/style issues.

* Refactored the codebase.

* Fix unused imports.

* Fix CSS file.

* Add new query to fetch chat history.

* Remove unnecessary console log.

* UI-UX improvements.

* Fix Css stuff.

* Fix fetch data.

* Remove unnecessary helper method.

* Final version.

* Fix empty-line missing.

* Additional improvements.

Co-authored-by: Renato Becker <renato.augusto.becker@gmail.com>

Siedlerchr committed:
[FIX] Add Authorization Bearer to allowed Headers (#8566)

ggazzo committed:
[FIX] Discussion sort option even with discussions disabled (#17963)

InstinctBas committed:
Encode custom oauth2 URL params (#13373)

Co-authored-by: Rodrigo Nascimento <rodrigoknascimento@gmail.com>

tonobo committed:
[NEW][API] Add `interation.update` endpoint (#13618)

Peym4n committed:
[NEW] Chatpal: add option to search in all public channels (#13856)

Peym4n committed:
[NEW] Chatpal: add option to search in all public channels (#13856)

ashwaniYDV committed:
[FIX] User can resend email verification if email is invalid or is empty (#16095)

pierre-lehnen-rc committed:
[NEW] Setting to determine if the LDAP user active state should be synced (#17645)

sampaiodiego committed:
Rename issue-close-app config file

gabriellsh committed:
[FIX] Cannot react while "Allow reaction" is set to true (#17964)

murtaza98 committed:
[FIX] Missing User when forwarding Omnichannel conversations via Apps-Engine (#17918)

sampaiodiego committed:
[FIX] Discussion not updating rooms list and not checking right permissions (#17959)

renatobecker committed:
[FIX] Autocomplete component is not working property when searching channels in the Livechat Departments form (#17970)

lpilz committed:
[IMPROVE] Slack import: Parse channel and user mentions (#17637)

* Treating channel mentions similarly to user mentions

* First iteration, links not clickable client-side yet.

* Made channel links clickable. #general import not working yet, as well as channel names with umlauts

* Finished implementation of clickable links. Had to add dname field to channel objects and message parser to make matching by parser possible without changing display of channels by using fname.

Co-authored-by: pierre-lehnen-rc <55164754+pierre-lehnen-rc@users.noreply.github.com>

stleitner committed:
[FIX] No rotate option, to prevent image quality loss (#15196)

c0dzilla committed:
channel's leave, hide options have channel in place of room (#10165)

pierre-lehnen-rc committed:
[IMPROVE] SAML implementation (#17742)

Montel committed:
[NEW][API] Endpoint `groups.setEncrypted` (#13477)

madhavmalhotra3089 committed:
[FIX] Update AmazonS3 file upload with error handling and sync operation (#10372)

localguru committed:
[NEW] Reply notification email to sender's email when the Direct Reply feature is disabled (#15767)

Co-authored-by: Diego Sampaio <chinello@gmail.com>

alexbartsch committed:
Make ldap avatar source field customizable (#12958)

fthiery committed:
[NEW] Ability to configure Jitsi room options via new setting `URL Suffix` (#17950)

gabriellsh committed:
[NEW] Rewrite Apps (#17906)

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>
Co-authored-by: Guilherme Gazzo <guilhermegazzo@gmail.com>

pierre-lehnen-rc committed:
[FIX] Admin UI is not rendering when trying to edit an user (#17972)

Co-authored-by: Guilherme Gazzo <guilhermegazzo@gmail.com>

ashwaniYDV committed:
[FIX] No Way to Display Password Policy on Password Reset Screen (#16400)

Co-authored-by: Guilherme Gazzo <guilhermegazzo@gmail.com>

Commit:
[NEW] Add ability to block failed login attempts by user and IP (#17783)

renatobecker committed:
[NEW] Allows agents to send chat transcript to omnichannel end-users (#17774)

hyfen committed:
Federation performance and bug fixes (#17504)

Commit:
Regression: App info broken (#17979)

Commit:
[IMPROVE] Refactor Omnichannel Office Hours feature (#17824)

* Change omnichannel office hours data structure

* Rename omnichannel office hours templates

* Renaming things from office hour to business hour

* Partially remove LivechatOfficeHour model

* remove obsolete setting

* Remove settings from the template

* Add support to cron jobs on omnichannel business hours

* Add dynamic template to omnichannel business hours

* Open and close business hours

* Remove references to LivechatOfficeHours

* Open and close business hour automatically on startup and setting changes

* Improve open and close business hours process

* Removing separated interfaces

* Apply suggestions from review

* Changing pt-br term

* General improvements on business hour layout

* Apply suggestions from review

* Remove online verification

* Change data structure

* Ignore if collection does not exist

Co-authored-by: Renato Becker <renato.augusto.becker@gmail.com>

ggazzo committed:
[IMPROVE] Threads (#17416)

Co-authored-by: Diego Sampaio <chinello@gmail.com>
Co-authored-by: gabriellsh <40830821+gabriellsh@users.noreply.github.com>
Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>

rodrigok committed:
Regression: Fix AWS S3 file retrieval (#17982)

bytes-by-talari committed:
[FIX] Add missing i18n entry for LDAP connection test success message (#17691)

sampaiodiego committed:
Regression: Only add reply-to if sender has emails (#17998)

pierre-lehnen-rc committed:
Regression: Image Upload not working (#17993)

pierre-lehnen-rc committed:
Regression: Admin User Edit panel is broken  (#17992)

pierre-lehnen-rc committed:
Regression Admin: uploading foreign profile image changes own image

renatobecker committed:
[FIX] New Omnichannel Past Chats list padding (#17994)

* Fix padding on the new design of the omnichannel past chats list.

* Add new improvements/fixes.

* Add missing blank line.

Commit:
[NEW][ENTERPRISE] Omnichannel multiple business hours (#17947)

* Change omnichannel office hours data structure

* Rename omnichannel office hours templates

* Renaming things from office hour to business hour

* Partially remove LivechatOfficeHour model

* remove obsolete setting

* Remove settings from the template

* Add support to cron jobs on omnichannel business hours

* Add dynamic template to omnichannel business hours

* Open and close business hours

* Remove references to LivechatOfficeHours

* Open and close business hour automatically on startup and setting changes

* Improve open and close business hours process

* Removing separated interfaces

* Apply suggestions from review

* Basic configuration to support multiple business hours

* Basic business hour insert and update operations

* Remove business hour

* Open multiple business hours automatically when the server starts

* Fix leak of business logic

* Open multiple business hours

* Close multiple business hour

* Changing pt-br term

* General improvements on business hour layout

* UI improvements

* Apply suggestions from review

* Add departments conditions

* Close business hours when remove department

* Remove online verification

* Refactor business hour to new data structure

* Change data structure

* Ignore if collection does not exist

* Fix businessHourManager startup.

* Remove console log.

* Fix Multiple business hours

* open/close business hours when removing/adding agent to department.

* Improve code

* Fix error on creating department.

* Add set to default when it is the last department

* Fix wrong condition

Co-authored-by: Renato Becker <renato.augusto.becker@gmail.com>

gabriellsh committed:
Regression: Cannot save avatar change on admin (#17999)

pierre-lehnen-rc committed:
[FIX] Missing i18n key for setting: Verify Email for External Accounts (#18002)

pierre-lehnen-rc committed:
[FIX] User is prompted to reset their password when logging in with OAuth provider (#18001)

tassoevan committed:
Remove useLazyRef hook usage (#18003)

pierre-lehnen-rc committed:
Regression - Incoming WebHook messages not showing up on the channel (#18005)

sampaiodiego committed:
Bump version to 3.4.0-rc.0

sampaiodiego committed:
Regression: Fix setting reply-to email header (#18008)

ggazzo committed:
Regression: Favorite and Featured fields not triggering changes (#18010)

sampaiodiego committed:
Regression: Deprecate check permission on integrations (#18024)

* Fix deprecate check permission on integrations

* Show warning once per hour

sampaiodiego committed:
Merge remote-tracking branch 'origin/develop' into release-candidate

sampaiodiego committed:
Bump version to 3.4.0-rc.1

pierre-lehnen-rc committed:
Regression - Unable to view or edit user status on the Edit User panel of the admin (#18032)

tassoevan committed:
Regression: Repair CodeMirror component reactivity (#18037)

MartinSchoeler committed:
Regression: Reset section button (#18007)

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>

ggazzo committed:
Regression: Room flickering if open a thread (#18004)

Co-authored-by: Diego Sampaio <chinello@gmail.com>

ggazzo committed:
Regression: Fix wrong message grouping inside threads (#18039)

sampaiodiego committed:
Merge remote-tracking branch 'origin/develop' into release-candidate

sampaiodiego committed:
Bump version to 3.4.0-rc.2

ggazzo committed:
Revert "Regression: Fix wrong message grouping inside threads (#18039)" (#18043)

This reverts commit 49183a83a76c9834466f40a32b4ab32f933cc82d.

ggazzo committed:
Regression: Grouping Thread messages (#18042)

pierre-lehnen-rc committed:
Regression: Improve the logic to get request IPs (#18033)

Co-authored-by: Diego Sampaio <chinello@gmail.com>

sampaiodiego committed:
Regression: Fix threads badge color indicators (#18048)

rodrigok committed:
Merge remote-tracking branch 'origin/develop' into release-candidate

rodrigok committed:
Bump version to 3.4.0-rc.3

Commit:
Regression: Improve Omnichannel Business Hours (#18050)

Co-authored-by: Renato Becker <renato.augusto.becker@gmail.com>

tassoevan committed:
Regression: Infinite render loop on Setup Wizard (#18074)

MartinSchoeler committed:
Regresion: Issue with reply button on broadcast channels (#18057)

ggazzo committed:
Regression: Fix exit-room on livechat (#18067)

Co-authored-by: Diego Sampaio <chinello@gmail.com>

renatobecker committed:
Upgrade Livechat Widget version to 1.6.0 (#18070)

ggazzo committed:
Regression: Fix mentions on thread preview (#18071)

MartinSchoeler committed:
Regression: Wrong padding and colors on some tabs (#18068)

Co-authored-by: Guilherme Gazzo <guilhermegazzo@gmail.com>

sampaiodiego committed:
Merge remote-tracking branch 'origin/develop' into release-candidate

sampaiodiego committed:
Bump version to 3.4.0-rc.4

sampaiodiego committed:
Regression: Fix update last message on delete (#18077)

sampaiodiego committed:
Merge remote-tracking branch 'origin/develop' into release-candidate

sampaiodiego committed:
Bump version to 3.4.0-rc.5

sampaiodiego committed:
Merge remote-tracking branch 'origin/release-candidate' into release-3.4.0

sampaiodiego committed:
Bump version to 3.4.0

sampaiodiego committed:
Merge pull request #18078 from RocketChat/release-3.4.0

Release 3.4.0

sampaiodiego committed:
Bump version to 3.5.0-develop

sampaiodiego committed:
Merge pull request #18083 from RocketChat/develop-sync

Merge master into develop & Set version to 3.5.0-develop

darigovresearch committed:
[FIX] Corrects Typo in Analytics section of the admin page (#17984)

gabriellsh committed:
[FIX] Geolocation permission being asked on load (#18030)

gabriellsh committed:
[FIX] Misleading labels in Prune Messages (#18006)

Co-authored-by: Guilherme Gazzo <guilhermegazzo@gmail.com>

MartinSchoeler committed:
[IMPROVE] Sort/Group list dropdown (#18081)

rodrigok committed:
[FIX] Not possible to read encrypted messages after disable E2E on channel level (#18101)

renatobecker committed:
Handle callback returning promise. (#18102)

gabriellsh committed:
[FIX] Avatar ETag missing from User (#18109)

paulobernardoaf committed:
[FIX] The livechat agent activity monitor wasn't being initialised because due to an internal error (#18090)

Densik committed:
[FIX] Email notifications were still being sent for online users (#18088)

ggazzo committed:
[FIX] Jitsi opening twice (#18111)

dudizilla committed:
[FIX] "Add reaction" icon missing when the viewport size is smaller than 500px (#18110)

MartinSchoeler committed:
[FIX] Clipboard not working when permalinking a pinned message (#18047)

gabriellsh committed:
[FIX] App details returns to apps table, instead of previous page (#18080)

omarchehab98 committed:
[FIX] Cannot open admin when server uses ROOT_URL with subpath (#18105) (#18147)

pierre-lehnen-rc committed:
[FIX] Old Data Migrations breaking upgrades (#18185)

d-gubert committed:
Update Apps-Engine version (#18212)

rodrigok committed:
[FIX] Application not loading due to reverse proxy decoding API calls unnecessarily (#18222)

pierre-lehnen-rc committed:
[FIX] SAML login crashing when receiving an array of roles (#18224)

pierre-lehnen-rc committed:
[FIX] CAS login not merging users with local accounts (#18238)

tassoevan committed:
Update the API of React Hooks using Meteor's reactive system (#18226)

tassoevan committed:
[FIX]Update link URL at AppsWhatIsIt (#18240)

tassoevan committed:
Regression: Remove calls to Console API in useForm hook (#18244)

ggazzo committed:
Rewrite Contextual Bar Discussion List in React (#18127)

ggazzo committed:
[IMPROVE] Message action styles (#18190)

djorkaeffalexandre committed:
[NEW] Sign in with apple (iOS client only) (#18258)

ogustavo-pereira committed:
[FEAT] Configurable character limit on live chat

* added new input for character limit setting

rodrigok committed:
[NEW] Added profile field to inform Nickname for users in order to be searchable (#18260)

gabriellsh committed:
Regression: Message actions under "unread messages" warning (#18273)

tassoevan committed:
[NEW] External MP3 encoder worker for audio recording (#18277)

gabriellsh committed:
[FIX] Apps page loading indefinitely if no Markeplace data (#18274)

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>
Co-authored-by: Guilherme Gazzo <guilherme@gazzo.xyz>

ggazzo committed:
[NEW] "My Account" rewritten (#18106)

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>
Co-authored-by: Gabriel Henriques <gabriel.henriques@rocket.chat>

ggazzo committed:
[FIX] Few adjustments to accept fuselage theme (#18009)

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>

pierre-lehnen-rc committed:
Regression - Profile page crashing for users without password  (#18287)

gabriellsh committed:
Regression: Account Sidebar not rendering properly (#18288)

ggazzo committed:
[FIX] Fix sticky notifications not working  (#18285)

d-gubert committed:
[FIX] View close uikit event sending wrong payload (#18289)

engelgabriel committed:
LingoHub based on develop (#18176)

* LingoHub Update :rocket:

Manual push by LingoHub User: Rodrigo Nascimento.
Project: Rocket.Chat

Made with :heart: by https://lingohub.com

Co-authored-by: Diego Sampaio <chinello@gmail.com>

gabriellsh committed:
Rewrite: My Account > Integrations rewritten (#18290)

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>

d-gubert committed:
[NEW] Update Apps-Engine version  (#18271)

ggazzo committed:
[NEW] User profile and User card (#18194)

* Handle Avatar on Error

* User Status Component

* User Card Component

* More variations

* More Variations

* Loading State

* useTimezoneTime Hook

* UTCClock Component

* Fix BaseAvatar

* getStatus Helper

* Rewrite User info to reuse + Stories

* Changes to open on Channels

* ???

* Fix reactAdapters

* Translation Provider

* Fix lint

* Improve loading state

* WIP

* UserCard and UserInfo Actions

* Lint

* Review

* wip

* Fix user info

* Update fuselage

* Update fuselage 0.13.0

* sad

* sad[2]

* fix camelCase paths

* Fix Ghost menu

* Fix some weird behaviors

* Fix missing user ID on actions

* Fix visual

* Fix VerticalBar

* Opss missing email field

* Missing some fields

* Update Fuselage

* Update app/theme/client/imports/components/contextual-bar.css

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>

* Remove Old userInfo template

* UserCard Api

* Update client/admin/customEmoji/CustomEmojiRoute.js

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>

* Remove unused variables

* Fix contextual-bar CSS selectors

* Destructure additional props in UserInfoWithData

* Remove obsolete props from VerticalBar

* Remove commented code

* Place useSetModal hook

* Apply suggestions from code review

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>

* Fiz markdown

* Update client/components/basic/UserCard.js

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>

* Update client/components/basic/avatar/UserAvatar.js

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>

* rename base avatar

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>
Co-authored-by: Gabriel Henriques <gabriel.henriques@rocket.chat>

d-gubert committed:
Update Apps-Engine to Beta version (#18294)

engelgabriel committed:
LingoHub based on develop (#18307)

* LingoHub Update :rocket:

Manual push by LingoHub User: Diego Sampaio.
Project: Rocket.Chat

Made with :heart: by https://lingohub.com

* Fix substitution variables

Co-authored-by: Diego Sampaio <chinello@gmail.com>

ggazzo committed:
[FIX] Closing the admin does not return to last opened room (#18308)

antkaz committed:
[FIX] Don't show agent info in the transcript if the setting is disabled (#18044)

Co-authored-by: Renato Becker <renato.augusto.becker@gmail.com>

gabriellsh committed:
[FIX] Delete user warning message undefined (#18310)

renatobecker committed:
[FIX][ENTERPRISE] Canned Responses not available to agents after they are added to departments

sampaiodiego committed:
[NEW][ENTERPRISE] Add support to license tags (#18093)

Co-authored-by: Guilherme Gazzo <guilhermegazzo@gmail.com>
Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>
Co-authored-by: Martin Schoeler <martin.schoeler@rocket.chat>

tobiasge committed:
[FIX] Merge user custom fields on LDAP sync (#17339)

gabriellsh committed:
[FIX] "Join" button on thread when room is read only (#18314)

renatobecker committed:
[FIX] Error when fetching a nonexistent business hour from the server (#18315)

rodrigok committed:
[IMPROVE] Mention autocomplete UI and performance improvements  (#18309)

Co-authored-by: Diego Sampaio <chinello@gmail.com>

pierre-lehnen-rc committed:
[FIX] Local Account login error when both LDAP and Email 2FA are enabled (#18318)

Co-authored-by: Diego Sampaio <chinello@gmail.com>

pierre-lehnen-rc committed:
[FIX] SAML login saves invalid username when receiving multiple values (#18213)

Co-authored-by: Diego Sampaio <chinello@gmail.com>

pierre-lehnen-rc committed:
[NEW][ENTERPRISE] Push Notification Data Privacy (#18254)

Co-authored-by: Rodrigo Nascimento <rodrigoknascimento@gmail.com>
Co-authored-by: Diego Sampaio <chinello@gmail.com>

Commit:
[IMPROVE] Improve performance and remove agents when the department is removed (#17049)

Co-authored-by: Renato Becker <renato.augusto.becker@gmail.com>

rodrigok committed:
Move the development guidelines to our handbook (#18026)

geekgonecrazy committed:
[FIX] Bug on entering token in connectivity services (#18317)

Co-authored-by: Diego Sampaio <chinello@gmail.com>

djorkaeffalexandre committed:
[IMPROVE] Change setting that blocks unauthenticated access to avatar to public (#18316)

Co-authored-by: Diego Sampaio <chinello@gmail.com>

sampaiodiego committed:
[FIX] SlackBridge error (#18320)

ggazzo committed:
Regression: Close UserCard if action opens a new page (#18319)

gabriellsh committed:
Regression: User Status selector (#18343)

gabriellsh committed:
Regression: Preferences crashing when User has no preferences set. (#18341)

dlassalle commented on app/livechat/server/methods/saveAppearance.js:
```diff
@@ -12,6 +12,7 @@ Meteor.methods({
 		const validSettings = [
 			'Livechat_title',
 			'Livechat_title_color',
+			'Livechat_title_character_limit',
```
Im having trouble getting this to save correctly. When I tried debugging the setting below...
```
		settings.forEach((setting) => {
			rcSettings.updateById(setting._id, setting.value);
                        console.log(`========DEBUG ${ setting._id } ${ setting.value }`.green)
                        console.log(rcSettings.get(setting._id))
		});
```
i got
```
I20200722-11:58:45.627(-4)? ========DEBUG Livechat_title Rocket.Chat
I20200722-11:58:45.627(-4)? Rocket.Chat
I20200722-11:58:45.632(-4)? ========DEBUG Livechat_title_color #366bc4
I20200722-11:58:45.632(-4)? #366bc4
I20200722-11:58:45.635(-4)? ========DEBUG Livechat_title_character_limit 13
I20200722-11:58:45.636(-4)? undefined
I20200722-11:58:45.638(-4)? ========DEBUG Livechat_show_agent_info true
I20200722-11:58:45.638(-4)? true
I20200722-11:58:45.640(-4)? ========DEBUG Livechat_show_agent_email true
I20200722-11:58:45.640(-4)? true
I20200722-11:58:45.642(-4)? ========DEBUG Livechat_display_offline_form false
I20200722-11:58:45.642(-4)? false
```

dlassalle commented on app/livechat/client/views/app/livechatAppearance.html:
```diff
@@ -16,6 +16,10 @@ <h2>{{_ "Settings"}}</h2>
 						<input type="color" class="preview-settings rc-input__element" name="color" id="color" value="{{color}}" />
 					</div>
+					<div class="input-line">
+						<label for="limitTextLength">{{_ "Characther Limit"}}</label>
```
```suggestion
						<label for="limitTextLength">{{_ "Character Limit"}}</label>
```

dlassalle reviewed (commented):
Ran into an issue when testing, can you have a look?
Also if you could cherry pick your two commits instead of the hundreds that our fork differs by, that would be great

MartinSchoeler committed:
Regression: Admin User password  (#18350)

graywolf336 committed:
[FIX] Update check not able to be disabled (#18339)

* Only run the update check when opting to register the server.
Doing it this way enables people to easily disable the update
check just like they can sending data to Rocket.Chat services.

* Add a new setting to enable/disable the update checker

Co-authored-by: Aaron Ogle <geekgonecrazy@users.noreply.github.com>

djorkaeffalexandre committed:
Regression: Notification with id-only isn't showed by iOS devices (#18353)

MartinSchoeler committed:
Regression: Userinfo center avatar image (#18354)

ggazzo committed:
Regression: Thread Title not being escaped (#18356)

gabriellsh committed:
Regression: nickname field in user profile. (#18359)

* Added nickname field, hints for disabled fields

* More missing nicknames

MartinSchoeler committed:
Fix broken link on readme (#18358)

sampaiodiego committed:
Regression: Fix defaultFields for null values (#18360)

ggazzo committed:
Regression: useStorage (#18370)

ggazzo committed:
Regression: MP3 worker (#18371)

gabriellsh committed:
Regression: Wrong background in disabled inputs  (#18372)

Co-authored-by: Tasso Evangelista <tasso.evangelista@rocket.chat>

gabriellsh committed:
Regression: Edit messages after opening thread (#18375)

ggazzo committed:
Regression: Mentions in thread title (#18369)

djorkaeffalexandre committed:
Regression: Provide a fallback text when push notification is idOnly (#18373)

Co-authored-by: Diego Sampaio <chinello@gmail.com>

sampaiodiego committed:
[FIX] Push gateway and cloud integration (#18377)

* Link push gateway setting to cloud register

* Add migration to disable gateway and alert admins

20051231 committed:
[FIX] File uploads for unknown file types but nothing is blocked (#18263)

ggazzo committed:
Regression: Fix useUserSubscription usage (#18378)

renatobecker committed:
Merge branch 'develop' into configurable-character-limit
